molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/controller/DataExplorerController.java

```diff
@@ -391,6 +399,31 @@ private boolean isGenomeBrowserEntity(EntityMetaData entityMetaData)
 		return attributeStartPosition != null && attributeChromosome != null;
 	}
 
+	/**
+	 * Updates the 'Entities' menu when an entity is deleted.
+	 * 
+	 * @param datasetName
+	 */
+	@RequestMapping(value = "/removeEntityFromMenu/{entityName}", method = POST)
+	public void updateMenuOnDeleteEntity(@PathVariable("entityName") String entityName, HttpServletResponse response)
+	{
+		Menu menu = menuManager.getMenu();
+		MenuItem mEntities = menu.findMenuItem("entities");
```

Comment:
Depending on the app, the form plugin for the requested entity might not be in this menu. Solution is to call menu.findMenuItem with the id of the form plugin for this entity.